django/core/validators.py

```diff
@@ -394,11 +394,35 @@ def compare(self, a, b):
 
 @deconstructible
 class StepValueValidator(BaseValidator):
-    message = _("Ensure this value is a multiple of step size %(limit_value)s.")
+    message = _(
+        "Please enter a valid value. The two nearest valid values are "
+        "%(nearest_lower)s and %(nearest_upper)s."
+    )
     code = "step_size"
 
+    def __init__(self, limit_value, message=None, offset=None):
+        super().__init__(limit_value, message)
+        self.offset = 0 if offset is None else offset
+
+    def __call__(self, value):
+        cleaned = self.clean(value)
+        limit_value = (
+            self.limit_value() if callable(self.limit_value) else self.limit_value
+        )
+        if self.compare(cleaned, limit_value):
+            frac, whole = math.modf(limit_value)
+            ndigits = len(str(frac)) - 2
+            remainder = cleaned.__class__(
+                abs(math.remainder(cleaned - self.offset, limit_value))
+            )
+            params = {
+                "nearest_lower": round(cleaned - remainder, ndigits),
+                "nearest_upper": round(cleaned - remainder + abs(limit_value), ndigits),
+            }
```

Comment:
Unfortunately, it is quite fragile :disappointed: I wonder if we could mention a list of valid values instead, e.g. _"Please enter a valid value. Valid values include 1.4, 4.4, 7.4, 10.4, and so on."_.

Reply:
@jrief What do you think? :point_up:

Reply:
Changed.